kcidb/test_oo.py

```diff
@@ -0,0 +1,960 @@
+"""kcidb.oo test module"""
+
+import kcidb
+from kcidb import db
+from kcidb import oo
+
+
+def list_of_data(data):
+    """
+
+    Args:
+        data:   An iterable object.
+
+    Returns:
+        A list of dictionary
+    """
+    return list(map(lambda x: x._data, data))
+
+
+class KCIDB00ClientTestCase(kcidb.unittest.TestCase):
+    """Test case for kcidb.oo module"""
+
+    def setUp(self):
+        """Database setup and oo client creation"""
+        self.source = db.Client('sqlite::memory:')
+
+        self.initialize_and_load_data()
+        self.assertTrue(isinstance(self.source, kcidb.orm.Source))
+
+        self.client = oo.Client(self.source)
+
+    def initialize_and_load_data(self):
+        """Function to load some known data into the database"""
+
+        if not self.source.is_initialized():
+            self.source.init()
+
+        self.source.load(
+            {
+                "version": {
+                    "major": 3,
```

Comment:
Could you create data with v4 schema instead? You can take your existing JSON data and just pass it through `kcidb-upgrade` to upgrade it to v4.